automation_tools/satellite6/upgrade/capsule.py

```diff
@@ -0,0 +1,161 @@
+        e.g '6.1','6.0'
+    TO_VERSION
+        Capsule version to upgrade to and enable repos while upgrading.
+        e.g '6.1','6.2'
+
+    """
+    sat_host = env.get('satellite_host')
+    from_version = os.environ.get('FROM_VERSION')
+    to_version = os.environ.get('TO_VERSION')
+    # Setup firewall rules on capsule node
+    setup_capsule_firewall()
+    # Setting Capsule61 Repos
+    major_ver = distro_info()[1]
+    # Re-register Capsule for 6.2
+    # AS per host unification feature: if there is a host registered where the
+    # Host and Content Host are in different organizations (e.g. host not in
+    # org, and content host in one), the content host will be unregistered as
+    # part of the upgrade process.
+    if to_version == '6.2':
+        ak_name = os.environ.get('CAPSULE_SUBSCRIPTION').split(',')[2].strip(
+            ) if os.environ.get('CAPSULE_SUBSCRIPTION') else os.environ.get(
+            'RHEV_CAPSULE_AK')
+        run('subscription-manager register --org="Default_Organization" '
+            '--activationkey={0} --force'.format(ak_name))
+    # if CDN Upgrade enable cdn repo
+    if os.environ.get('CAPSULE_URL') is None:
+        enable_repos('rhel-{0}-server-satellite-capsule-{1}-rpms'.format(
+            major_ver, to_version))
+    disable_repos('rhel-{0}-server-satellite-capsule-{1}-rpms'.format(
+        major_ver, from_version))
+    if from_version == '6.1' and major_ver == '6':
+        enable_repos('rhel-server-rhscl-{0}-rpms'.format(major_ver))
+    if from_version == '6.0':
+        # Stop katello services, except mongod
+        run('for i in qpidd pulp_workers pulp_celerybeat '
+            'pulp_resource_manager httpd; do service $i stop; done')
+    run('yum clean all', warn_only=True)
+    print('Wait till packages update ... ')
+    print('YUM UPDATE started at: {0}'.format(time.ctime()))
+    update_packages(quiet=False)
+    print('YUM UPDATE finished at: {0}'.format(time.ctime()))
+    if from_version == '6.0':
+        run('yum install -y capsule-installer', warn_only=True)
+        # Copy answer file from katello to capule installer
+        run('cp /etc/katello-installer/answers.capsule-installer.yaml.rpmsave '
+            '/etc/capsule-installer/answers.capsule-installer.yaml',
+            warn_only=True)
+    # Generates Capsule Certs file on satelltie and copies in capsule
+    execute(
+        generate_capsule_certs,
+        cap_host,
+        True,
+        host=sat_host
+    )
+    # Copying the capsule cert to capsule
+    execute(lambda: run("scp -o 'StrictHostKeyChecking no' {0}-certs.tar "
+                        "root@{0}:/home/".format(cap_host)), host=sat_host)
+    # Rebooting the system to see possible errors
+    if os.environ.get('RHEV_CAP_HOST'):
+        reboot(120)
+        if to_version == '6.1':
+            # Stopping the services again which started in reboot
+            run('for i in qpidd pulp_workers pulp_celerybeat '
```

Comment:
Actually this services stop is happening on satellite 6.0 only which is before upgrading to 6.1 and so the service stopping procedure will remain same. So just to avoid the confusion I changed if block as if from_version == '6.0'.